ducc/lib/image.go

```diff
@@ -483,7 +483,56 @@ func getManifestWithUsernameAndPassword(img *Image, user, pass string) ([]byte,
 	return body, nil
 }
 
+type Credentials struct {
+	username string
+	password string
+}
+
+func getCredentialsFromEnv(user, pass string) (Credentials, error) {
+	u := os.Getenv(user)
+	p := os.Getenv(pass)
+	c := Credentials{u, p}
+	err := error(nil)
+	if user == "" || pass == "" {
+		err = fmt.Errorf("missing either username ($%s) or password ($%s) or both for accessing the docker registry", user, pass)
+	}
+	return c, err
+
+}
+
+func getDockerHubCredentials() (Credentials, error) {
+	return getCredentialsFromEnv("DUCC_DOCKERHUB_USER", "DUCC_DOCKERHUB_PASS")
+}
+
+func getGitlabContainersCredentials() (Credentials, error) {
+	return getCredentialsFromEnv("DUCC_GITLAB_USER", "DUCC_GITLAB_PASS")
+}
+
+func GetAuthToken(url string, credentials []Credentials) (token string, err error) {
+	docker, err := getDockerHubCredentials()
+	if err == nil {
+		credentials = append(credentials, docker)
+	}
+	gitlab, err := getGitlabContainersCredentials()
+	if err == nil {
+		credentials = append(credentials, gitlab)
+	}
+	for _, c := range credentials {
```

Comment:
If I understand correctly, we would probably try with wrong credentials to one of the services.  Not sure if this is harmful.  Does it make sense to look at the URL to decide whether to use Docker or Gitlab credentials?